core/commonMain/src/implementations/immutableMap/PersistentHashMap.kt

```diff
@@ -63,6 +64,14 @@ internal class PersistentHashMap<K, V>(internal val node: TrieNode<K, V>,
     }
 
     override fun putAll(m: Map<out K, @UnsafeVariance V>): PersistentMap<K, V> {
+        if (m is PersistentHashMap) @Suppress("UNCHECKED_CAST") {
```

Comment:
Let's optimize `PersistentHashMapBuilder.putAll` and use it here as it was done in `PersistentHashSet.addAll `.